src/main/scala/com/johnsnowlabs/ml/tensorflow/TensorResources.scala

```diff
@@ -12,24 +14,40 @@ class TensorResources {
   private val tensors = ArrayBuffer[Tensor[_]]()
 
   def createTensor[T](obj: T): Tensor[_] = {
-    val result = if (obj.isInstanceOf[String]) {
-      Tensor.create(obj.asInstanceOf[String].getBytes("UTF-8"), classOf[String])
-    }
-    else {
-      Tensor.create(obj)
+    val result = obj match {
+      case str: String =>
+        Tensor.create(str.getBytes("UTF-8"), classOf[String])
+      case _ =>
+        Tensor.create(obj)
     }
 
     tensors.append(result)
     result
   }
 
+  def createIntBufferTensor[T](shape: Array[Long], buf: IntBuffer): Tensor[_] = {
+
+    val result = Tensor.create(shape, buf)
```

Comment:
Is there any chance the Tensor we are creating is already on the buffer?

Reply:
No, the data we are sending to the previous `Tensor.create()` was object and TF decides the shape and the rest by itself. That is why it's less optimized and it is proffered to use `Tensor.create(shape, buffer)`.